airbyte-server/src/main/java/io/airbyte/server/handlers/JobHistoryHandler.java

```diff
@@ -122,7 +143,15 @@ public JobDebugInfoRead getJobDebugInfo(final JobIdRequestBody jobIdRequestBody)
     final Job job = jobPersistence.getJob(jobIdRequestBody.getId());
     final JobInfoRead jobinfoRead = jobConverter.getJobInfoRead(job);
 
-    return buildJobDebugInfoRead(jobinfoRead);
+    final JobDebugInfoRead jobDebugInfoRead = buildJobDebugInfoRead(jobinfoRead);
+    if (temporalClient != null) {
+      final UUID connectionId = UUID.fromString(job.getScope());
+      temporalClient.getWorkflowState(connectionId)
```

Comment:
@gosusnp will this break if the _next_ job has started? basically for connection X:
* job 1 happens and completes. 
* then job 2 starts. 
* while job 2 is running i look up the debug info for job 1. won't it say running (because this `getWorkflowState` is keyed only by connection id), but i would expect it to not be running.

Reply:
It think what you described is correct since workflowState is per connection id and has no history.

The other option that was considered was to have a different API endpoint, I think this felt like a smaller lift at that time to remove some noise from the tests.

Reply:
got it. we should at least have a todo or warning in the code then. that's going to catch someone by surprise. i didn't figure it out until i was in the shower a couple hours later. 😅